### PR TITLE
Emit declaration file

### PR DESCRIPTION
**Pull request by nickserv**
Should hopefully fix issues with intellisense and TypeScript in dependent projects, which are caused by types not being published

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,12 +49,14 @@
     ]
   },
   "dependencies": {
+    "@types/lodash": "^4.14.121",
```
Comment (jabacchetta) [inline package.json:52]: From testing locally, it doesn't look like adding these to dependencies provides any additional benefit, but maybe I'm missing something?

**Review by jabacchetta — CHANGES_REQUESTED** [review]
Thanks for your help on this!

From testing locally, it does look like this partially solves IntelliSense. It does provide IntelliSense for first-level parameters. However, it doesn't provide IntelliSense for options objects. And the imports still aren't showing up. Looks like I'll need to continue to investigate. Working and non-working examples shown below:

**Import working with lodash:**

import-lodash

**Import not working with coconut:**

import-coconut

**Options object IntelliSense working in the package itself:**

package-options

**Options object IntelliSense not working on project with package installed:**

project-options

**Comment by jabacchetta** [comment]
There is a chance that the import IntelliSense is just a project configuration issue (i.e. `jsconfig.json`). Although this seems to be the only package I've had issues with.

Additionally, with first-level parameter IntelliSense working, it's strange that options object IntelliSense isn't working, and that wouldn't seem to be a project configuration issue.

File: package.json (hunk 1 above)
Comment (nickserv) [reply package.json:52]: The reasoning is that by installing packages with types to dependencies, dependent packages using TypeScript won't have to reinstall them and can share the package version (as having two different versions of the same type package can cause type errors).

Source: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies

Personally I prefer to put type packages in the same category as the original package. Also TypeScript's docs show TypeScript itself in dependencies, the rationale is that while it is a development tool, it still includes types for production code.

**Comment by nickserv** [comment]
> From testing locally, it does look like this partially solves IntelliSense. It does provide IntelliSense for first-level parameters. However, it doesn't provide IntelliSense for options objects. And the imports still aren't showing up. Looks like I'll need to continue to investigate. Working and non-working examples shown below:

So if I'm understanding correctly it works internally, but not in external packages? How have you tried to test your package? I'd recommend trying both `npm link` (which is more convenient) and `npm pack` with `npm install` (solves some issues with type package duplication when packages are linked with `npm link`). 

> Additionally, with first-level parameter IntelliSense working, it's strange that options object IntelliSense isn't working, and that wouldn't seem to be a project configuration issue.

I'm guessing that the top level declaration file is correct, but the other types aren't being concatenated into the output declaration file with `outFile` for some reason. I'd recommend trying `outDir` instead, I was almost going to use it originally but I didn't notice this issue.

File: package.json (hunk 1 above)
Comment (jabacchetta) [reply package.json:52]: Interesting, thanks for the explanation!

**Pull request merged**